sdk/communication/Shared/src/CommunicationIdentifierSerializerTest.cs

```diff
@@ -172,5 +172,45 @@ public void DeserializeMicrosoftTeamsUser(bool isAnonymous)
             Assert.AreEqual(expectedIdentifier.RawId, identifier.RawId);
             Assert.AreEqual(expectedIdentifier, identifier);
         }
+
+        [Test]
+        public void DeserializeMicrosoftTeamsUser_WithKind_DeserializesSuccessfully()
```

Comment:
can you pls add tests for all kinds? you never know what can go wrong :)

Reply:
Done!